### PR TITLE
Expose extra frontend options through build apk/ios/macOS

### LINKED ISSUE #53174: Enable the usage of  --delete-tostring-package-uri in the flutter tool
These are extra frontend options that need to be passed to the frontend server invocation as a multi option, such as `--delete-tostring-package-uri=dart:ui` and `--delete-tostring-package-uri=package:flutter`.

We could make these options the default in the future without changing the behavior here, or expose delete-tostring directly, though users might have trouble configuring it.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

This will allow experimenting with the remove to string transformer before we're ready to turn it on by default. This doesn't work for web yet since we use dart2js instead of the frontend_server for producing kernel

Fixes https://github.com/flutter/flutter/issues/53174

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/build_bundle.dart [hunk 1]
```diff
@@ -49,15 +50,7 @@ class BuildBundleCommand extends BuildSubCommand {
...
       )
-      ..addMultiOption(FlutterOptions.kExtraFrontEndOptions,
-        splitCommas: true,
-        hide: true,
-      )
       ..addOption('asset-dir', defaultsTo: getAssetBuildDirectory())
-      ..addMultiOption(FlutterOptions.kExtraGenSnapshotOptions,
-        splitCommas: true,
-        hide: true,
-      )
```
Comment (dnfield) [inline packages/flutter_tools/lib/src/commands/build_bundle.dart]: Intentional?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/build_bundle.dart]: Ahh yeah, but I'll do argument cleanup another time

**Review by dnfield — APPROVED** [review]
LGTM

**Comment by dnfield** [comment]
...once the CI issues are fixed :)

**Comment by jonahwilliams** [comment]
:D

**Pull request merged**